### ISSUE TITLE
[1.2.0] [Linux] Segmentation Fault when we choose the Exit Option in the Pause Menu

### ISSUE THREAD

**Issue by JackoboLeChocobo** [Rubberduckycooly/Sonic-1-2-2013-Decompilation#162]
Hi,

I meet this bug during my test with Sonic 1. The culprit is in Debug.cpp line 385 :
```
    RemoveNativeObjectType(StartMenu_Create, StartMenu_Main);
    CreateNativeObject(StartMenu_Create, StartMenu_Main);
```

There is an issue with CreateNativeObject or RemoveNativeObjectType function that creates in this case badly the StartMenu_Main var.
Finish the game normally and access main menu in Dev Menu doesn't crash the game.

Tested on LUbuntu 20.04 64-bits.

**Comment by JackoboLeChocobo** [comment]
Temp fix - for the moment it works : 
```
	RemoveNativeObjectType(StartMenu_Create, StartMenu_Main);
	RemoveNativeObjectType(PauseMenu_Create, PauseMenu_Main);
       
    if (mode == 0 || !GetGlobalVariableByName("timeAttack.result")) {
		    CreateNativeObject(StartMenu_Create, StartMenu_Main);
		    CreateNativeObject(StartMenu_Create, StartMenu_Main);
		    CreateNativeObject(StartMenu_Create, StartMenu_Main);
		    CreateNativeObject(StartMenu_Create, StartMenu_Main);
    }
    else {
    }
```

**Comment by MGRich** [comment]
To be fixed in 1.2.0 (if it isn't fixed already.)
